### DISCUSSION TITLE
For LFV meson decays why quark flavors are not specified separately?

### DISCUSSION THREAD

**Discussion by Joro20** [flav-io/flavio#262] [Q&A]
For LFV B meson decays I find in the 'bpll_lfv.py' file there is no specification of different flavor of quarks! Although in the file it is mentioned that for Lepton flavour violating decays
`def _define_obs_B_Mll(M, ll):`
      `_process_tex = _hadr_lfv[M]['tex']+' '+_tex_lfv[''.join(ll)]`
     ` _process_taxonomy = r'Process :: $b$ hadron decays :: FCNC decays :: $B\to P\ell^+\ell^-$ :: $' 
     + _process_tex + r"$"`
    ` _obs_name = "BR("+M+''.join(ll)+")"`
 etc. 

I'm not sure how is it referring to different quark flavors for decays such as B-Ke\mu which is at quark level 'b->s' transition! Is it picking up that information from the `_hadr_lfv[M]` which is defined as

`_hadr_lfv = {
'B0->K': {'tex': r"\bar B^0\to \bar K^0", 'B': 'B0', 'P': 'K0', },
'B+->K': {'tex': r"B^-\to K^-", 'B': 'B+', 'P': 'K+', },
'B0->pi': {'tex': r"\bar B^0\to \pi^0", 'B': 'B0', 'P': 'pi0', },
'B+->pi': {'tex': r"B^-\to \pi^-", 'B': 'B+', 'P': 'pi+', },
}`  ?

If that is the case, then how this information is translated to the Wilson coefficients with different  quark flavors ?

**Comment by DavidMStraub** [comment]
The function that does that is here
https://github.com/flav-io/flavio/blob/master/flavio/physics/bdecays/wilsoncoefficients.py#L154

**Reply by Joro20** [reply]
Thank you @DavidMStraub for pointing towards the exact location. Can you please let me know, from where is it getting the information about `B` and `P` ? Are those lists of B meson and Pseudoscalar mesons defined in 
https://github.com/flav-io/flavio/blob/v2.6.1/flavio/physics/bdecays/common.py
?
But how to choose a particular set of `B` and `P` ? Also what is the role of `M` in the function `def _define_obs_B_Mll(M, ll)` ?

**Reply by DavidMStraub** [reply]
Sorry but if you want to understand exactly how the source code works, you need to look at the source code.

**Reply by Joro20** [reply]
Indeed I'm trying to understand how things works so that if I need to calculate some new observable I can have a workable code by myself! By 'source code' are you referring to understanding of Python language ? That might be problematic as I'm not familiar with the language! Instead I've invited you to look at the file (in my repository) with the particular observable that I want to calculate. If possible please let me know what I'm doing wrong. Thanks.

**Reply by DavidMStraub** [reply]
Flavio is a Python library...
